onnxscript/rewriter/__init__.py

```diff
@@ -17,37 +17,50 @@
 from onnxscript.ir.passes.common import unused_removal
 from onnxscript.rewriter import pattern
 
-RewriteRuleSet = pattern.RewriteRuleSet
 PatternRewriteRule = pattern.RewriteRule
 
 ModelProtoOrIr = TypeVar("ModelProtoOrIr", onnx.ModelProto, ir.Model)
 
 
+class RewritePass(ir.passes.InPlacePass):
+    def __init__(
+        self,
+        pattern_rewrite_rules: Sequence[PatternRewriteRule] | pattern.RewriteRuleSet = (),
+    ) -> None:
+        if pattern_rewrite_rules:
+            if not isinstance(pattern_rewrite_rules, pattern.RewriteRuleSet):
+                # Create a pattern rule-set using provided rules
+                pattern_rewrite_rules = pattern.RewriteRuleSet(pattern_rewrite_rules)
+        assert isinstance(pattern_rewrite_rules, pattern.RewriteRuleSet)
+        self.pattern_rewrite_rules: pattern.RewriteRuleSet = pattern_rewrite_rules
+
+    def call(self, model: ir.Model) -> ir.passes.PassResult:
+        count = self.pattern_rewrite_rules.apply_to_model(model)
+        if count:
+            print(f"Applied {count} of general pattern rewrite rules.")
+        return ir.passes.PassResult(model, bool(count))
+
+
 def rewrite(
     model: ModelProtoOrIr,
-    pattern_rewrite_rules: Union[Sequence[PatternRewriteRule], RewriteRuleSet] = (),
+    pattern_rewrite_rules: Union[Sequence[PatternRewriteRule], pattern.RewriteRuleSet] = (),
 ) -> ModelProtoOrIr:
     if isinstance(model, onnx.ModelProto):
         model_ir = ir.serde.deserialize_model(model)
         proto = True
     else:
         model_ir = model
         proto = False
-    if pattern_rewrite_rules:
-        if not isinstance(pattern_rewrite_rules, RewriteRuleSet):
-            # Create a pattern rule-set using provided rules
-            pattern_rewrite_rules = pattern.RewriteRuleSet(pattern_rewrite_rules)
-        count = pattern_rewrite_rules.apply_to_model(model_ir)
-        if count:
-            print(f"Applied {count} of general pattern rewrite rules.")
-    unused_remover = ir.passes.PassManager(
+
+    rewrite_pass = ir.passes.PassManager(
```

Comment:
I think this is causing a regression when no rewrite-rules are specified. I think we should retain an `if pattern_rewrite_rules` condition and avoid the RewritePass if no rules are specified.